File: Ahorn/entities/FlameGeyser.jl
module FlameGeyser

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/FlameGeyser" FlameGeyser(x::Integer, y::Integer, damageRadius::Integer=30, eruptDuration::Number=1.0, eruptInterval::Number=4.0, flameHeight::Integer=200, warningTime::Number=1.0)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(FlameGeyser)
)

function Ahorn.selection(entity::FlameGeyser)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::FlameGeyser, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: Loenn/entities/flameGeyser.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local flameGeyser = {}

flameGeyser.name = "MaggyHelper/FlameGeyser"
flameGeyser.depth = -50
flameGeyser.fieldInformation = {
    eruptInterval = {
        minimumValue = 0
    },
    eruptDuration = {
        minimumValue = 0
    },
    warningTime = {
        minimumValue = 0
    },
    flameHeight = {
        minimumValue = 0
    },
    damageRadius = {
        minimumValue = 0
    }
}
flameGeyser.placements = {
    {
        name = "normal",
        data = {
            eruptInterval = 4.0,
            eruptDuration = 1.0,
            warningTime = 1.0,
            flameHeight = 200,
            damageRadius = 30
        }
    }
}

function flameGeyser.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/flame_geyser/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function flameGeyser.selection(room, entity)
    return utils.rectangle(entity.x - 16, entity.y - 16, 32, 16)
end

return flameGeyser
